#! /bin/sh -e
## kill-runaway-crded_tool_debianization.dpatch by Andres Mejia <mcitadel@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to modify Tools/LinuxScripts/kill-runaway-crded script for use in a Debian based distro.
## This patch will work provided that the source tree was checked out to a directory named alienarena2007.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- Tools/LinuxScripts/kill-runaway-crded~	2007-05-13 01:23:35.000000000 -0400
+++ Tools/LinuxScripts/kill-runaway-crded	2007-05-13 01:23:35.000000000 -0400
@@ -1,9 +1,9 @@
 #!/bin/sh
 
-for TPPID in `ps -eo pid,pcpu,comm | grep crded | grep -v grep|
-awk '{print $1":"$2}' |
-awk 'BEGIN {FS=":"}{ if ($2 > 20.0){print $1} }'`
+### This script has been patched for distribution to Debian systems.
+
+for TPPID in `ps -eo pid,comm | grep crded | cut -d' ' -f1`
 do
 /bin/kill -9 $TPPID
+rm -f `ls $HOME/.alienarena2007/alienarena2007-server.*.pid`
 done
-
